src/FakeItEasy/Core/MethodInfoManager.cs

```diff
@@ -160,7 +160,7 @@ private static bool TypeImplementsInterface(Type type, Type interfaceType)
             return type.GetInterfaces().Any(x => x.Equals(interfaceType));
         }
 
-        private struct TypeMethodInfoPair
+        private struct TypeMethodInfoPair : IEquatable<TypeMethodInfoPair>

```

Comment:
I don't remember the warning number, but apparently you should implement `IEquatable<T>` when you override `Equals`.